File: Data Structures & Algorithms/two-integer-sum-ii/submission-13.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        prev = None

        def binarySearch(value, l):
            r = len(numbers)

            while l < r:
                m = (l + r) // 2

                if numbers[m] < value: l = m + 1
                elif value < numbers[m]: r = m
                else: return m
            
            return None
            


        for i, num in enumerate(numbers):
            if prev and prev == num: continue


            remainder = target - num
            if remainder >= num:
                index = binarySearch(remainder, i + 1)
                if index: return [i + 1, index + 1]


            prev = num

File: Data Structures & Algorithms/binary-search/submission-13.java
class Solution {
    public int search(int[] nums, int target) {
        List<Integer> numList = Arrays.stream(nums).boxed().collect(Collectors.toList());
        Collections.sort(numList);
        int index = Collections.binarySearch(numList, target);

        return index >= 0 ? index : -1;
    }
}
